File: solutions/medium/382.Linked_List_Random_Node/382.Linked_List_Random_Node.py
# Definition for singly-linked list.
import random
from typing import Optional


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:

    def __init__(self, head: Optional[ListNode]):
        self.range = []
        while head:
            self.range.append(head.val)
            head = head.next

    def getRandom(self) -> int:
        pick = int(random.random() * len(self.range))
        return self.range[pick]


# Your Solution object will be instantiated and called as such:
# obj = Solution(head)
# param_1 = obj.getRandom()

File: solutions/medium/382.Linked_List_Random_Node/382.Linked_List_Random_Node.ts

  // Definition for singly-linked list.
  class ListNode {
      val: number
      next: ListNode | null
      constructor(val?: number, next?: ListNode | null) {
          this.val = (val===undefined ? 0 : val)
          this.next = (next===undefined ? null : next)
      }
  }


class Solution {
  nodeSeen:number[] = []

  constructor(head: ListNode | null) {
      while(head != null) {
          this.nodeSeen.push(head.val)
          head = head.next
      }
  }

  getRandom(): number {
      const index = Math.floor(Math.random() * this.nodeSeen.length)
    return this.nodeSeen[index]
  }
}

/**
* Your Solution object will be instantiated and called as such:
* var obj = new Solution(head)
* var param_1 = obj.getRandom()
*/